rules/TypeDeclaration/Rector/ClassMethod/ReturnTypeFromStrictReturnExprRector.php

```diff
@@ -80,12 +80,26 @@ public function refactor(Node $node): ?Node
         return $node;
     }
 
+    public function provideMinPhpVersion(): int
+    {
+        return PhpVersion::PHP_74;
+    }
+
     private function hasSingleStrictReturn(ClassMethod $classMethod): bool
     {
         if ($classMethod->stmts === null) {
             return false;
         }
 
+        if ($this->betterNodeFinder->hasInstancesOf($classMethod->stmts, [Yield_::class])) {
```

Comment:
should exclude inner function, there is `hasInstancesOfInFunctionLikeScoped` method for it:

```suggestion
        if ($this->betterNodeFinder->hasInstancesOfInFunctionLikeScoped($classMethod, [Yield_::class])) {
```